docker-up.sh

```diff
@@ -89,38 +90,38 @@ while [ "$1" ]; do
     shift
 done
 
-docker-compose --version
+docker compose version
 
 if [ $(($(date +%s) - $(date -r docker-compose.yml +%s))) -gt 604800 ]; then
-    # pull updates weekly
+    # Pull updates weekly
     echo "Checking for updates to the service images..."
     update
     echo
 fi
 
 mkdir -p /tmp/aplus
 trap onexit INT
-docker-compose up & pid=$!
+docker compose up & pid=$!
 
-help_n=4 # show first info after 24 seconds
+help_n=4 # Show first info after 24 seconds
 while kill -0 $pid 2>/dev/null; do
-    while read -rs -t 0; do read -rs -t 0.1; done # flush input
-    read -rsn1 -t 6 i # read a byte
+    while read -rs -t 0; do read -rs -t 0.1; done # Flush input
+    read -rsn1 -t 6 i # Read a byte
     # (1 or 142) -> timeout (6s). Show help every 50 times (every 5 minutes)
     [[ $? != 0 ]] && { ((--help_n > 0)) && continue || help_n=50; }
     case "$i" in
         q|Q) break ;;
         s|S) keep="x" ; break ;;
-        $'\e') # escape (ESC or ANSI code)
-            read -rsn1 -t 0.01 i # try to read a second byte
-            [ $? -eq 142 ] && { keep="x"; break; } # timeout -> no second byte -> plain ESC
+        $'\e') # Escape (ESC or ANSI code)
+            read -rsn1 -t 0.01 i # Try to read a second byte
+            [ $? -eq 142 ] && { keep="x"; break; } # Timeout -> no second byte -> plain ESC
             ;;
     esac
 
-    # print status and help
+    # Print status and help
     echo
     echo "  List of alive containers:"
-    { docker container ls --filter "name=^${COMPOSE_PROJECT_NAME}_"  --format "{{.ID}}" | xargs docker container inspect --format '	{{.Name}}	{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}	{{range $p, $conf := .NetworkSettings.Ports}}{{$p}} {{end}}'; } 2>/dev/null || true
+    { docker container ls --filter "name=^${COMPOSE_PROJECT_NAME}-"  --format "{{.ID}}" | xargs docker container inspect --format '	{{.Name}}	{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}	{{range $p, $conf := .NetworkSettings.Ports}}{{$p}} {{end}}'; } 2>/dev/null || true
```

Comment:
Is it now guaranteed that the container names always use dashes (like `aplus-plus-1`) instead of underscores (`aplus_plus_1`)? Or could it still vary between platforms like Linux and Windows?

Reply:
After a quick experiment with the old and new Compose versions, I think that the old version uses underscores (`_`) in the container names and the new version uses hyphens (`-`). At least on my Linux, I witnessed such a difference.

```
# New
docker compose version
Docker Compose version v2.21.0

# Old
docker-compose --version
docker-compose version 1.29.2, build unknown
```

That's enough about the underscore. No need to spend more time on that.